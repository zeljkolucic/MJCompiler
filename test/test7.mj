program MJProgram

int a, b, c;
int niz[], niz2[];

char c1, c2, c3;
{

	void main(){
		niz = new int[3];
		niz2 = new int[4];
		
		a = len(niz) + len(niz2);
		
		if(a == 7)
			print(1);
		else
			print(-1);
			
		c1 = 'b';
		b = ord(c1);
		c = ord('y');
		if(c - b == 23)
			print(1);
		else
			print(-1);
			
		c2 = chr(b);
		c3 = chr(c);
		
		if(c2 == 'b' && c3 == 'y')
			print(1);
		else
			print(-1);
	}

}