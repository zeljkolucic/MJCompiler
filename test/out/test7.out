Linija [10]: Pristup globalnoj promenljivoj niz.
Linija [11]: Pristup globalnoj promenljivoj niz2.
Linija [13]: Pristup globalnoj promenljivoj a.
Linija [13]: Pristup globalnoj promenljivoj niz.
Linija [13]: Poziv funkcije len.
Linija [13]: Pristup globalnoj promenljivoj niz2.
Linija [13]: Poziv funkcije len.
Linija [15]: Pristup globalnoj promenljivoj a.
Linija [20]: Pristup globalnoj promenljivoj c1.
Linija [21]: Pristup globalnoj promenljivoj b.
Linija [21]: Pristup globalnoj promenljivoj c1.
Linija [21]: Poziv funkcije ord.
Linija [22]: Pristup globalnoj promenljivoj c.
Linija [22]: Poziv funkcije ord.
Linija [23]: Pristup globalnoj promenljivoj c.
Linija [23]: Pristup globalnoj promenljivoj b.
Linija [28]: Pristup globalnoj promenljivoj c2.
Linija [28]: Pristup globalnoj promenljivoj b.
Linija [28]: Poziv funkcije chr.
Linija [29]: Pristup globalnoj promenljivoj c3.
Linija [29]: Pristup globalnoj promenljivoj c.
Linija [29]: Poziv funkcije chr.
Linija [31]: Pristup globalnoj promenljivoj c2.
Linija [31]: Pristup globalnoj promenljivoj c3.
=====================SYMBOL TABLE DUMP=========================
Type int: int, -1, -1 
Type char: char, -1, -1 
Con eol: char, 10, 0 
Con null: Class [], 0, 0 
Meth chr: char, 0, 1 
   Var i: int, 0, 1 

Meth ord: int, 0, 1 
   Var ch: char, 0, 1 

Meth len: int, 0, 1 
   Var arr: Arr of notype, 0, 1 

Type bool: , -1, -1 
Prog MJProgram: notype, 0, 1 
   Var a: int, 0, 0 
   Var b: int, 1, 0 
   Var c: int, 2, 0 
   Var niz: Arr of int, 3, 0 
   Var niz2: Arr of int, 4, 0 
   Var c1: char, 5, 0 
   Var c2: char, 6, 0 
   Var c3: char, 7, 0 
   Meth main: notype, 0, 0 



